test/src/unit-capi-subarray_2.cc

```diff
@@ -0,0 +1,3411 @@
+/**
```

Comment:
Ordinarily, if we properly deprecated the old way of setting subarray ranges to the query, we would remove from the tests the old way and we would add tests only for the new APIs. I can see the argument of keeping the tests even for the deprecated APIs, since they are still exposed to the users. But just adding a note here to remember to delete the older file when we remove the old APIs.

Reply:
Nothing done for this yet.
Open to suggestions for where to make such a note that it might actually be seen when those APIs are removed...?

Reply:
created https://app.clubhouse.io/tiledb-inc/story/9201/when-appropriate-remove-supplanted-tiledb-query-apis-and-related-items

Reply:
I suggest we don't add this file folks and just replace the old deprecated API with the new APIs. If the user is using the old APIs with this version, they really need to switch to the new ones.